### ISSUE TITLE
Report BLEU in training mode

### ISSUE THREAD

**Issue by haorannlp** [pytorch/fairseq#2877]
### Before asking:   
1. search the issues.   
2. search the docs.    

#### What is your question?
I'm training a Transformer and now I want to report the BLEU score on a small training set during training using greedy decoding or sampling decoding. Are there any commands for this functionality? If not, could you give me some advice on how to modify the source code? Thank you so much!
#### What's your environment?

 - fairseq Version (e.g., 1.0 or master):
 - PyTorch Version (e.g., 1.0)
 - OS (e.g., Linux):
 - How you installed fairseq (`pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:
